python/ray/data/read_api.py

```diff
@@ -698,6 +703,155 @@ def read_parquet(
+            implementation to read from. These filesystems are specified in the
+            `pyarrow docs <https://arrow.apache.org/docs/python/api/\
+            filesystems.html#filesystem-implementations>`_. Specify this parameter if
+            you need to provide specific configurations to the filesystem. By default,
+            the filesystem is automatically selected based on the scheme of the paths.
+            For example, if the path begins with ``s3://``, the ``S3FileSystem`` is
+            used. If ``None``, this function uses a system-chosen implementation.
+        columns: A list of column names to read. Only the specified columns are
+            read during the file scan.
+        parallelism: The amount of parallelism to use for the dataset. Defaults to -1,
+            which automatically determines the optimal parallelism for your
+            configuration. You should not need to manually set this value in most cases.
+            For details on how the parallelism is automatically determined and guidance
+            on how to tune it, see :ref:`Tuning read parallelism
+            <read_parallelism>`. Parallelism is upper bounded by the total number of
+            records in all the parquet files.
+        ray_remote_args: kwargs passed to :meth:`~ray.remote` in the read tasks.
+        tensor_column_schema: A dict of column name to PyArrow dtype and shape
+            mappings for converting a Parquet column containing serialized
+            tensors (ndarrays) as their elements to PyArrow tensors. This function
+            assumes that the tensors are serialized in the raw
+            NumPy array format in C-contiguous order (e.g., via
+            `arr.tobytes()`).
+        meta_provider: A :ref:`file metadata provider <metadata_provider>`. Custom
+            metadata providers may be able to resolve file metadata more quickly and/or
+            accurately. In most cases you do not need to set this parameter.
+        arrow_parquet_args: Other parquet read options to pass to PyArrow. For the full
+            set of arguments, see the`PyArrow API <https://arrow.apache.org/docs/\
+                python/generated/pyarrow.dataset.Scanner.html\
+                    #pyarrow.dataset.Scanner.from_fragment>`_
+
+    Returns:
+        :class:`~ray.data.Dataset` producing records read from the specified parquet
+        files.
+    """
+    if is_in_databricks_runtime():
+        spark = get_spark_session()
+        paths = spark.read.load(uri, format="delta").inputFiles()
+        paths = [_convert_dbfs_path_to_local_path(path) for path in paths]
+    else:
+        try:
+            from deltalake import DeltaTable
+        except ImportError as e:
+            raise Exception("Please install deltalake to use read_delta.") from e
+
+        # TODO: check the limitations of DeltaTable and params each version supports
+        dt = DeltaTable(uri, version, storage_options)
+        paths = dt.file_uris()
```

Comment:
Alternative:

```
        spark = SparkSession.builder.config(
            "spark.jars.packages", "io.delta:delta-core_2.12:2.2.0"
        )
        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
        .config(
            "spark.sql.catalog.spark_catalog",
            "org.apache.spark.sql.delta.catalog.DeltaCatalog",
        )
        .getOrCreate()
        spark.read.load(uri, format="delta").inputFiles()
        ...
```

Reply:
Q:

For apache/spark:

assuming a spark session has been created without delta extension config,
then execute
```
        spark = SparkSession.builder.config(
            "spark.jars.packages", "io.delta:delta-core_2.12:2.2.0"
        )
        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
        .config(
            "spark.sql.catalog.spark_catalog",
            "org.apache.spark.sql.delta.catalog.DeltaCatalog",
        )
        .getOrCreate()
```

will the returned spark session work with delta dataset ?

Reply:
If the spark is created already, the configs will get applied to the existing spark session. I'll have a test, but one thing needs to be noted here is the version of io.delta:delta-core_2.12, it depends on spark version. We might need a mapping to load different jar package version.

Reply:
<img width="692" alt="image" src="https://github.com/ray-project/ray/assets/82044803/6b473240-37cd-4fac-9449-326fb17e5607">
<img width="664" alt="image" src="https://github.com/ray-project/ray/assets/82044803/1f8bec28-2b7e-4410-a7ad-10c99e6c5b42">

It works fine

Reply:
Added new unit test for this.